synapseclient/core/exceptions.py

```diff
@@ -124,36 +149,126 @@ def _raise_for_status(response, verbose=False):
+                message += f"\n{BODY_PREFIX}{response.text}\n\n"
             except:  # noqa
-                message += "\nCould not append all response info"
+                message += f"\n{UNABLE_TO_APPEND_RESPONSE}"
+
+        raise SynapseHTTPError(message, response=response)
+
+
+def _raise_for_status_httpx(
+    response: httpx.Response, logger: logging.Logger, verbose: bool = False
+) -> None:
+    """
+    Replacement for requests.response.raise_for_status().
+    Catches and wraps any Synapse-specific HTTP errors with appropriate text with the
+    HTTPX library in mind.
+
+    Arguments:
+        response: The response object from the HTTPX request.
+        verbose: If True, the request and response information will be appended to the
+            error message.
+    """
+
+    message = None
+
+    # TODO: Add more meaningful Synapse-specific messages to each error code
+    # TODO: For some status codes, throw other types of exceptions
+    if 400 <= response.status_code < 500:
+        # TODOs:
+        # 400: 'bad_request'
+        # 401: 'unauthorized'
+        # 402: 'payment_required'
+        # 403: 'forbidden'
+        # 404: 'not_found'
+        # 405: 'method_not_allowed'
+        # 406: 'not_acceptable'
+        # 407: 'proxy_authentication_required'
+        # 408: 'request_timeout'
+        # 409: 'conflict'
+        # 410: 'gone'
+        # 411: 'length_required'
+        # 412: 'precondition_failed'
+        # 413: 'request_entity_too_large'
+        # 414: 'request_uri_too_large'
+        # 415: 'unsupported_media_type'
+        # 416: 'requested_range_not_satisfiable'
+        # 417: 'expectation_failed'
+        # 418: 'im_a_teapot'
+        # 422: 'unprocessable_entity'
+        # 423: 'locked'
+        # 424: 'failed_dependency'
+        # 425: 'unordered_collection'
+        # 426: 'upgrade_required'
+        # 428: 'precondition_required'
+        # 429: 'too_many_requests'
+        # 431: 'header_fields_too_large'
+        # 444: 'no_response'
+        # 449: 'retry_with'
+        # 450: 'blocked_by_windows_parental_controls'
+        # 451: 'unavailable_for_legal_reasons'
+        # 499: 'client_closed_request'
+        message_body = _get_message(response, logger)
+        message = f"{response.status_code} {CLIENT_ERROR} {message_body}"
+
+    elif 500 <= response.status_code < 600:
+        # TODOs:
+        # 500: 'internal_server_error'
+        # 501: 'not_implemented'
+        # 502: 'bad_gateway'
+        # 503: 'service_unavailable'
+        # 504: 'gateway_timeout'
+        # 505: 'http_version_not_supported'
+        # 506: 'variant_also_negotiates'
+        # 507: 'insufficient_storage'
+        # 509: 'bandwidth_limit_exceeded'
+        # 510: 'not_extended'
+        message_body = _get_message(response, logger)
+        message = f"{response.status_code} {SERVER_ERROR} {message_body}"
+
+    if message is not None:
+        if verbose:
```

Comment:
nit: Do we want to remove the if-nesting and just have `if verbose and message is not None`

Reply:
Not in this case.

`if message is not None` - We'll `raise SynapseHTTPError(message, response=response)`

However,

`if verbose` - We'll append the request/response information to the message.